web-modules/javax-servlets-2/src/test/java/com/baeldung/setparam/SanitizeParametersRequestWrapperUnitTest.java

```diff
@@ -54,7 +54,7 @@ public void whenSetValueToWrapperParametersStringArray_thenThe2ndCallShouldNotEq
 
         firstCallValues[0] = NEW_VALUE;
         String[] secondCallValues = wrapper.getParameterValues("input");
-        assertNotEquals(firstCallValues, secondCallValues);
+        assertThat(firstCallValues).isNotEqualTo(secondCallValues);
```

Comment:
I believe @KevinGilmore might have been trying to suggest you to do something like this:
```suggestion
        assertThat(secondCallValues).doesNotContain(NEW_VALUE);
```

Reply:
isNotEqualTo is correct.
We are asserting whether the 2nd call of getParaemterValues() is not equal to the modified value from the 1st call.

Reply:
I understand that isNotEqualTo correctly asserts that the 2nd call is not equal to the modified value from the 1st call

However, this is not really using any of the power of AssertJ to make the intention of the assertion clearer, which is what I think Kevin was trying to get you to take a look at

Logically, you are modifying with NEW_VALUE and then you want to assert that this change was not effective

A buggy implementation might for example append NEW_VALUE to the end of the array, and your isNotEqualTo assertion would be green even though there is a bug

By explicitly checking for NEW_VALUE being absent, you get a bit more test coverage, and the intention of the assertion is clearer

Reply:
Assertion updated